bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingRegistryImpl.java

```diff
@@ -124,23 +124,23 @@ public void removeThingTracker(ThingTracker thingTracker) {
 
     @Override
     protected void notifyListenersAboutAddedElement(Thing element) {
-        super.notifyListenersAboutAddedElement(element);
         postEvent(ThingEventFactory.createAddedEvent(element));
         notifyTrackers(element, ThingTrackerEvent.THING_ADDED);
+        super.notifyListenersAboutAddedElement(element);
     }
 
     @Override
     protected void notifyListenersAboutRemovedElement(Thing element) {
-        super.notifyListenersAboutRemovedElement(element);
```

Comment:
I think for removing a thing the order was already fine. If we remove the thing from the ThingManager before notifying listeners, they could try to disable the thing in that short gap.
Moreover, if we leave the order in this method unchanged, then it will be exactly the reverse order from the `notifyListenersAboutAddedElement` method, which is also what I would expect.

Reply:
That is correct. I will update the PR.